Cargo.toml

```diff
@@ -39,6 +39,7 @@ hex = "0.4.0"
 hmac = "~0.7.1"
 lazy_static = "1.4.0"
 md-5 = "0.8.0"
+num = "0.3.0"
```

Comment:
This dependency adds functionality to compare numbers of different types that might result in lossy conversion (as required [here](https://github.com/mongodb/specifications/blob/master/source/unified-test-format/unified-test-format.rst#flexible-numeric-comparisons)).

Reply:
Is just just needed for tests? If so, we can put it in `dev-dependencies` so that cargo can skip downloading/compiling it for users of the driver.